app/controllers/distributions_controller.rb

```diff
@@ -44,31 +42,25 @@ def index
 
   def create
     @distribution = Distribution.new(distribution_params.merge(organization: current_organization))
+    @storage_locations = current_organization.storage_locations
 
     if @distribution.valid?
-      if params[:commit] == "Preview Distribution"
-        @distribution.line_items.combine!
-        @line_items = @distribution.line_items
-        render :show
+      @distribution.storage_location.decrease_inventory @distribution
+
+      if @distribution.save
+        update_request(params[:distribution][:request_attributes], @distribution.id)
+
+        send_notification(current_organization, @distribution)
+        flash[:notice] = "Distribution created!"
+        session[:created_distribution_id] = @distribution.id
+        redirect_to distributions_path
       else
-        @distribution.storage_location.distribute!(@distribution)
-
-        if @distribution.save
-          update_request(params[:distribution][:request_attributes], @distribution.id)
-
-          send_notification(current_organization.id, @distribution.id)
-          flash[:notice] = "Distribution created!"
-          session[:created_distribution_id] = @distribution.id
-          redirect_to distributions_path
-        else
-          flash[:error] = "There was an error, try again?"
-          render :new
-        end
+        flash[:error] = "There was an error, try again?"
+        render :new
       end
     else
-      @storage_locations = current_organization.storage_locations
       flash[:error] = "An error occurred, try again?"
-      logger.error "failed to save distribution: #{@distribution.errors.full_messages}"
+      logger.error "[!] DistributionsController#create failed to save distribution: #{@distribution.errors.full_messages}"
```

Comment:
I know we use the rails logger in a couple places. I wonder if it would make sense to have a custom log file since (I think) the logger just writes to production.log which is probably easy to miss stuff in.
(Obviously not part of this issue)

Reply:
That's actually a really good idea - like `ActivityLogger` or something -- custom outputs and all that. It would definitely make it a lot easier to find user errors!

Reply:
Would it be possible to put a helper method somewhere that outputs this kind of log entry to _both_ the current and a new special log? It would be helpful for the error to be in the current log as well, right?

Reply:
We should make an issue for that. I think it's not in-scope for this change (a heavy refactor on storage location I/O), but I definitely agree with a lot of what you're saying and we should improve our logging methods

Reply:
#1002